From 0c7b704097409cc6500284c51fada0b4f3a695eb Mon Sep 17 00:00:00 2001
Message-Id: <0c7b704097409cc6500284c51fada0b4f3a695eb.1607538352.git.edvin.torok@citrix.com>
From: Jonathan Davies <jonathan.davies@citrix.com>
Date: Fri, 7 Apr 2017 14:27:18 +0100
Subject: oxenstored: initialise logging earlier

Otherwise we miss out on messages from things that try to log earlier in
the start-up procedure.

Signed-off-by: Jonathan Davies <jonathan.davies@citrix.com>
Reviewed-by: Wei Liu <wei.liu2@citrix.com>
Reviewed-by: Christian Lindig <christian.lindig@citrix.com>
Release-acked-by: Julien Grall <julien.grall@arm.com>

diff --git a/tools/ocaml/xenstored/xenstored.ml b/tools/ocaml/xenstored/xenstored.ml
index 0f62a23a17..0b56f81f17 100644
--- a/tools/ocaml/xenstored/xenstored.ml
+++ b/tools/ocaml/xenstored/xenstored.ml
@@ -288,6 +288,8 @@ let _ =
 
 	let quit = ref false in
 
+	Logging.init_xenstored_log();
+
 	if cf.restart then (
 		DB.from_file store domains cons "/var/run/xenstored/db";
 		Event.bind_dom_exc_virq eventchn
@@ -317,7 +319,6 @@ let _ =
 	Sys.set_signal Sys.sigusr1 (Sys.Signal_handle (fun _ -> sigusr1_handler store));
 	Sys.set_signal Sys.sigpipe Sys.Signal_ignore;
 
-	Logging.init_xenstored_log();
 	if cf.activate_access_log then begin
 		let post_rotate () = DB.to_file store cons "/var/run/xenstored/db" in
 		Logging.init_access_log post_rotate
